### PR TITLE
feat: Implement _mm_shuffle_pi16

### PR DESCRIPTION
**Pull request by HowJMay**

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -2357,6 +2357,77 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
     return vreinterpretq_m128_f32(vdupq_n_f32(0));
 }
 
+FORCE_INLINE __m64 _mm_shuffle_pi16(__m64 a, int imm)
```
Comment (jserv) [inline sse2neon.h:2360]: Add Intel intrinsic descriptions.

File: sse2neon.h [hunk 2]
```diff
@@ -2357,6 +2357,77 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
...
+{
+    int16x4_t ret = vdup_n_s16(0);
+    int16_t a0 = vget_lane_s16(vreinterpret_s16_m64(a), 0);
+    int16_t a1 = vget_lane_s16(vreinterpret_s16_m64(a), 1);
+    int16_t a2 = vget_lane_s16(vreinterpret_s16_m64(a), 2);
+    int16_t a3 = vget_lane_s16(vreinterpret_s16_m64(a), 3);
+
+    switch (imm & 0x3) {
```
Comment (jserv) [inline sse2neon.h:2368]: If `__builtin_shufflevector` is available, use `__builtin_shufflevector` to implement this intrinsic.

Comment (HowJMay) [reply sse2neon.h:2377]: added

File: sse2neon.h (hunk 1 above)
Comment (HowJMay) [reply sse2neon.h]: done

File: sse2neon.h [hunk 3]
```diff
@@ -2357,6 +2357,90 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
...
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_shuffle_pi16
+#if __has_builtin(__builtin_shufflevector)
+#define _mm_shuffle_pi16(a, imm)                                           \
+    __extension__({                                                        \
+        vreinterpret_m64_s16(__builtin_shufflevector(                      \
+            vreinterpret_s16_m64(a), vreinterpret_s16_m64(a), (imm & 0x3), \
+            ((imm >> 2) & 0x3), ((imm >> 4) & 0x3), ((imm >> 6) & 0x3)));  \
+    })
+#else
+FORCE_INLINE __m64 _mm_shuffle_pi16(__m64 a, int imm)
```
Comment (jserv) [inline sse2neon.h:2371]: Would you re-implement this intrinsic in the form of macro? Such as `_mm_shuffle_ps_default`.

Comment (HowJMay) [reply sse2neon.h]: done

File: sse2neon.h [hunk 4]
```diff
@@ -2357,6 +2357,36 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
...
+            vget_lane_s16(vreinterpret_s16_m64(a), ((imm) >> 4) & 0x3), ret,   \
+            2);                                                                \
+        ret = vset_lane_s16(                                                   \
+            vget_lane_s16(vreinterpret_s16_m64(a), ((imm) >> 6) & 0x3), ret,   \
+            3);                                                                \
+        vreinterpret_m64_s16(ret);                                             \
+    })
+#endif
+
+
```
Comment (jserv) [inline sse2neon.h:2389]: Keep one blank line.

Comment (jserv) [reply sse2neon.h:2389]: I was not aware of the above style change.

Comment (HowJMay) [reply sse2neon.h]: done

**Pull request merged**